packages/logger/src/index.ts

```diff
@@ -0,0 +1,151 @@
+import isError = require('lodash.iserror');
+import util = require('util');
+
+type Level = 'debug' | 'error' | 'fatal' | 'info' | 'trace' | 'warn';
+
+type LoggerConstructorOptions = {
+  asyncOperationId?: string,
+  console?: Console,
+  executions?: string,
+  granules?: string,
+  parentArn?: string,
+  pretty?: boolean,
+  sender?: string,
+  stackName?: string,
+  version?: string
+};
+
+class Logger {
+  readonly #asyncOperationId: string | undefined;
+  readonly #executions: string | undefined;
+  readonly #granules: string | undefined;
+  readonly #parentArn: string | undefined;
+  readonly #pretty: boolean;
+  readonly #sender: string;
+  readonly #stackName: string | undefined;
+  readonly #console: Console;
+  readonly #version: string | undefined;
+
+  constructor(options: LoggerConstructorOptions = {}) {
+    this.#asyncOperationId = options.asyncOperationId;
+    this.#executions = options.executions;
+    this.#granules = options.granules;
+    this.#parentArn = options.parentArn;
+    this.#pretty = options.pretty || false;
+    this.#sender = options.sender || 'unknown';
+    this.#stackName = options.stackName;
+    this.#console = options.console || global.console;
+    this.#version = options.version;
+  }
+
+  /**
+   * Log a debug message
+   */
+  debug(...messageArgs: any[]) {
+    this.writeLogEvent('debug', messageArgs);
+  }
+
+  /**
+   * Log an error message
+   */
+  error(...messageArgs: any[]) {
+    const lastMessageArg = messageArgs[messageArgs.length - 1];
+
+    if (isError(lastMessageArg)) {
+      const error = lastMessageArg;
+
+      let actualMessageArgs = messageArgs.slice(0, messageArgs.length - 1);
+      if (actualMessageArgs.length === 0) actualMessageArgs = [error.message];
+
+      const additionalKeys: { error: { name: string, message: string, stack?: string[] }} = {
+        error: {
+          name: error.name,
+          message: error.message
+        }
+      };
+      if (error.stack) additionalKeys.error.stack = error.stack.split('\n');
+
+      this.writeLogEvent(
+        'error',
+        actualMessageArgs,
+        additionalKeys
+      );
+    } else {
+      this.writeLogEvent('error', messageArgs);
+    }
+  }
+
+  /**
+   * Log a fatal message
+   */
+  fatal(...messageArgs: any[]) {
```

Comment:
Why is this set as any?  The interfacing looks similar to the other non-error methods, including the map-through from common/log 🤔

Reply:
They should all actually be `any` because [that is what `format` accepts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/util.d.ts#L8). I'll fix the inconsistency.